### PR TITLE
[M1595] Bleaching Validators

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[Port Bleaching validations to v2](https://trello.com/c/rlLM2Bey/1595-port-bleaching-validations-to-v2)

File: src/api/submission/validations2/validators/generic.py [hunk 1]
```diff
@@ -75,3 +63,72 @@ def __call__(self, collect_record, **kwargs):
...
+
+        duplicates = [r for r in duplicate_tracker.values() if len(r) > 1]
+        if not duplicates:
+            return OK
+
+        return ERROR, self.DUPLICATE_VALUES, {"duplicates": duplicates}
+
+
+class PositiveIntegerValidator(BaseValidator):
+    NOT_POSITIVE_INTEGER = "not_postitive_integer"
```
Comment (saanobhaai) [inline src/api/submission/validations2/validators/generic.py:104]: spelling

File: src/api/submission/validations2/validators/quadrat_size.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        self.quadrat_size_path = quadrat_size_path
+        super().__init__(**kwargs)
+
+    @validator_result
+    def __call__(self, collect_record, **kwargs):
+        quadrat_size = self.get_value(collect_record, self.quadrat_size_path)
+
+        try:
+            quadrat_size = float(quadrat_size)
+            if quadrat_size == 0:
```
Comment (saanobhaai) [inline src/api/submission/validations2/validators/quadrat_size.py:18]: `quadrat_size <= 0`?

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Damned impressive.

File: src/api/submission/validations2/validators/__init__.py [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 from .base import ERROR, IGNORE, OK, WARN, BaseValidator, ValidatorResult
 from .biomass import BiomassValidator
+from .bleaching_percent import BleachingPercentValidator
+from .colony_count import ColonyCountValidator
 from .depth import DepthValidator
 from .fish_count import FishCountValidator, TotalFishCountValidator
 from .fish_family import FishFamilySubsetValidator
 from .fish_size import FishSizeValidator
-from .generic import AllEqualValidator, ListRequiredValidator, RequiredValidator
+from .generic import AllEqualValidator, DuplicateValidator, ListPositiveIntegerValidator, ListRequiredValidator, PositiveIntegerValidator, RequiredValidator
```
[the remark this answers is not in the document]
Comment (gridcell) [reply src/api/submission/validations2/validators/__init__.py:10]: I prefer being explicit so I know what's being imported and works better with ide

**Pull request merged**